lib/terminus.js

```diff
@@ -46,21 +49,23 @@ const intialState = {
 function noop () {}
 
 function decorateWithHealthCheck (server, state, options) {
-  const { healthChecks, logger } = options
+  const { healthChecks, logger, onSendFailureDuringShutdown } = options
 
   server.listeners('request').forEach((listener) => {
     server.removeListener('request', listener)
     server.on('request', async (req, res) => {
       if (healthChecks[req.url]) {
         if (state.isShuttingDown) {
-          return sendFailure(res)
+          await sendFailure(res, null, onSendFailureDuringShutdown)
+          return
         }
         let info
         try {
           info = await healthChecks[req.url]()
         } catch (error) {
           logger('healthcheck failed', error)
-          return sendFailure(res, error.causes)
+          await sendFailure(res, error.causes)
```

Comment:
can we simply `return await sendFailure`?

Reply:
Ah yes, we could. It'll have to be `return sendFailure` rather than `return await sendFailure` though, see here for more info:
https://eslint.org/docs/rules/no-return-await